evalml/tests/demo_tests/test_datasets.py

```diff
@@ -86,5 +86,31 @@ def test_datasets(dataset_name, expected_shape, local_datasets):
 def test_datasets_match_local(dataset_name, demo_method, local_datasets):
     X, y = demo_method
     X_local, y_local = local_datasets[dataset_name]
+
+    if dataset_name == "daily_temp":
+        missing_date_1 = pd.DataFrame([pd.to_datetime("1984-12-31")], columns=["Date"])
+        missing_date_2 = pd.DataFrame([pd.to_datetime("1988-12-31")], columns=["Date"])
+        missing_y_1 = pd.Series([14.5], name="Temp")
+        missing_y_2 = pd.Series([14.5], name="Temp")
+
+        X_local = pd.concat(
+            [
+                X_local.iloc[:1460],
+                missing_date_1,
+                X_local.iloc[1460:2920],
+                missing_date_2,
+                X_local.iloc[2920:],
+            ]
+        ).reset_index(drop=True)
+        y_local = pd.concat(
+            [
+                y_local.iloc[:1460],
+                missing_y_1,
+                y_local.iloc[1460:2920],
+                missing_y_2,
+                y_local.iloc[2920:],
+            ]
+        ).reset_index(drop=True)
```

Comment:
Shouldn't this already be changed with this PR's `.load_weather()` function modifications?

Reply:
This is changing the `X_local` and `y_local` variables which are read in from the `local_datasets`. They come from `conftest` via `load_data` instead of `load_weather`, so they have to be changed again. I decided to move this modification into that `conftest` function instead to make it look cleaner!